paasta_tools/kubernetes/bin/paasta_secrets_sync.py

```diff
@@ -447,8 +574,6 @@ def sync_crypto_secrets(
         if not secret_data:
             continue
 
-        # In order to prevent slamming the k8s API, add some artificial delay here
-        time.sleep(0.3)
```

Comment:
@nemacysts note I refactored this out and added an associated time.sleep mock in the tests to consider it. didn't make sense to call it anywhere but the function. in the future, the function can be throttled using something similar to a <rate_limit> annotation eg https://pypi.org/project/ratelimit/